pkg/git/git.go

```diff
@@ -23,6 +23,32 @@ const (
 	gitProtocol   = "ssh"
 )
 
+// ExtractHostnamePort extracts the hostname and port of the provided Git URL
+func ExtractHostnamePort(url string) (string, int, error) {
+	endpoint, err := transport.NewEndpoint(url)
+	if err != nil {
+		return "", 0, err
+	}
+
+	port := endpoint.Port
+
+	if port == 0 {
+		switch endpoint.Protocol {
+		case httpProtocol:
+			port = 80
+		case httpsProtocol:
+			port = 443
+		case gitProtocol:
+			port = 22
+
+		default:
+			return "", 0, fmt.Errorf("Unknown protocol: %s", endpoint.Protocol)
```

Comment:
Nit: As per https://go.dev/wiki/Errors, errors should not start with capital letter.
```suggestion
			return "", 0, fmt.Errorf("unknown protocol: %s", endpoint.Protocol)
```